### PR TITLE
CFEP-11 Automated Closing of Excessively Old PRs on Staged Recipes

### PR DESCRIPTION
**Pull request by beckermr**
This CFEP proposes a simple form of staged-recipes automation that would nicely close excessively old PRs. 

It is related to #9, but is a lot simpler and so hopefully we can implement it now.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge/staged-recipes @conda-forge/core for viz

**Comment by CJ-Wright** [comment]
Any thoughts on implementation? Should we be using the stale bot?

**Comment by beckermr** [comment]
I was going to make a cron job in circle CI but yeah we can use whatever service we like. The circleCI cron job would be easy for the staged-recipes team to maintain since it would only need the current github API keys and changes to the circleCI config. I am not familiar with the stale bot.

**Comment by CJ-Wright** [comment]
https://github.com/apps/stale
https://probot.github.io/apps/stale/

A cron job may be simpler, although one needs to be careful about handling the GH API limits.

**Comment by beckermr** [comment]
Yeah, TBH though the stale bot there looks even simpler than a cron job since I won't need to grep the github API and compare dates. Good tip!

**Comment by beckermr** [comment]
Though it does not appear to work on pull requests.

**Comment by CJ-Wright** [comment]
Does it not? The Readme has `A GitHub App built with Probot that closes abandoned Issues and Pull Requests after a period of inactivity.`

**Comment by beckermr** [comment]
Ahhh ok! Great! Then stale bot it is!

**Comment by beckermr** [comment]
Alright, the CFEP is updated to say we are using the stale bot.

**Comment by CJ-Wright** [comment]
If/when you are ready for a vote, you can post a comment with the info from [here](https://conda-forge.org/docs/orga/governance.html#voting)

**Comment by beckermr** [comment]
@conda-forge/core

This PR falls under the CFEP Approval policy, please vote and/or comment on this PR.
This PR needs 60% of core to vote yea to pass.
To vote please leave Approve (yea) or Request Changes (nay) reviews.
If you would like changes to the current language please leave a comment or push to this branch.
This vote will end on 2019-11-22.

**Comment by beckermr** [comment]
@CJ-Wright I very likely made the wrong comment above. Please help!

**Review by scopatz — CHANGES_REQUESTED** [review]
I am not a fan of using bots to automate human interaction

**Comment by beckermr** [comment]
Even for PRs that are 6 months old?

**Comment by mbargull** [comment]
> I am not a fan of using bots to automate human interaction

I understand the concern, but I'd like us to come up with some pragmatic solution (technically but also user perception-wise) ;).
@scopatz, if not fully automated, what are your thoughts on semi-automation, i.e., let the bot only do some but not all steps of commenting, labelling, and closing a PR and then let a human do the rest? If that seems acceptable, which part do you think is the most important one to do by a human? (I'd assume commenting, but I can't really tell.)

**Comment by scopatz** [comment]
Yeah, I think the commenting and closing should be done by a human. Labeling / commenting that it a PR is at-risk of closure can be done by a bot, if desired.

**Comment by beckermr** [comment]
Hmmmm. OK. At literally hundreds of PRs that are excessively old, this is quite a lot of manual labor to clean out. I'd personally rather not be responsible for such labor, so I'd like to either fully automate this process, or simply leave things as they are.

**Comment by beckermr** [comment]
Just to add some numbers, there are 289 PRs that are from 2018 or earlier.

**Comment by beckermr** [comment]
One possible side effect is that this would effect issues too. We may not want that, but I am not opposed.

**Comment by beckermr** [comment]
Ahhhh we can exclude labels, like the package request label etc.

**Comment by beckermr** [comment]
Bump here @conda-forge/core. 

This cfep is to add the stale bot to staged recipes.

**Comment by pkgw** [comment]
Does the stale bot have an option to make this a two-stage process, where first there's a warning of imminent closure and then the PR is actually closed like a week later? I don't think there's any way to know for sure, but I feel like people would feel a bit better about having some warning before a bot closes their PRs.

If this feature isn't easily available, I'm still in favor. I feel that if we write a message that is very explicit "this was just a robot, go ahead and reopen this PR right away if you feel like it", it won't be alienating.

**Comment by beckermr** [comment]
Yes it does two stages

**Review by pkgw — APPROVED** [review]

**Review by mariusvniekerk — APPROVED** [review]

File: cfep-11.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+
+<table>
+<tr><td> Title </td><td> Simple staged-recipes Automation </td>
```
Comment (isuruf) [inline cfep-11.md]: IMO, the title doesn't reflect what is being proposed.

Comment (beckermr) [reply cfep-11.md]: Sure! Happy to change!

Comment (beckermr) [reply cfep-11.md]: I changed the title @isuruf. LMK if you think this is better!

**Review by isuruf — APPROVED** [review]

**Review by xhochy — APPROVED** [review]

**Review by ocefpaf — APPROVED** [review]

**Review by mwcraig — APPROVED** [review]
We've had a bot closing stale PRs for a year or two in the main astropy repo and it has worked pretty well. It posts a warning that the PR will close in a month if there are no changes (or a special label applied). The best side effect it seems to have is that it reminds me to look at PRs I had forgotten about.

**Comment by beckermr** [comment]
@CJ-Wright This CFEP needs 4 or 5 more votes by my count, but the voting period has officially ended. What is the procedure here?

**Review by ericdill — APPROVED** [review]

**Comment by mbargull** [comment]
> voting period has officially ended. What is the procedure here?

We've been rather lenient on this, so I wouldn't worry about it.

(And I just noticed that I hadn't approved yet, which I thought I already did, whoops..)

**Review by mbargull — APPROVED** [review]

**Review by jakirkham — APPROVED** [review]

**Review by djsutherland — APPROVED** [review]

**Review by loriab — APPROVED** [review]

**Comment by beckermr** [comment]
Just one more here before it passes IIUIC.

**Review by 183amir — APPROVED** [review]

File: cfep-11.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+
+<table>
+<tr><td> Title </td><td> Automated Closing of Excessively Old PRs on Staged Recipes </td>
+<tr><td> Status </td><td> Draft  </td></tr>
+<tr><td> Author(s) </td><td> Matthew R Becker &lt;becker.mr@gmail.com&gt; C.J. Wright &lt;&gt;</td></tr>
```
Comment (CJ-Wright) [inline cfep-11.md]:
```suggestion
<tr><td> Author(s) </td><td> Matthew R Becker &lt;becker.mr@gmail.com&gt; Christopher J. "CJ" Wright &lt;cjwright4242@gmail.com&gt;</td></tr>
```

**Review by soapy1 — APPROVED** [review]

**Comment by beckermr** [comment]
Huzzzzah! Thanks @soapy1! It appears that we have passed this CFEP!

**Comment by CJ-Wright** [comment]
Can you update this to `accepted`? Then I can merge (or I can merge as is and we can fix it later)

**Pull request merged**